internal/static/integrations/ssh_exporter/ssh_client.go

```diff
@@ -0,0 +1,194 @@
+package ssh_exporter
+
+import (
+    "bytes"
+    "fmt"
+    "os"
+    "path/filepath"
+    "time"
+    "os/exec"
+    "strings"
+
+    "github.com/go-kit/log"
+    "github.com/go-kit/log/level"
+    "golang.org/x/crypto/ssh"
+    "golang.org/x/crypto/ssh/knownhosts"
+    "os/user"
+)
+
+
+type SSHClient struct {
+    config  *ssh.ClientConfig
+    host    string
+    port    int
+    logger  log.Logger
+    timeout time.Duration
+}
+var sshKeyscanCommand = func(targetAddress string) ([]byte, error) {
+    cmd := exec.Command("ssh-keyscan", "-H", targetAddress)
+    return cmd.Output()
+}
+
+func ensureKnownHosts(knownHostsPath, targetAddress string) error {
+    // Ensure .ssh directory exists
+    if err := os.MkdirAll(filepath.Dir(knownHostsPath), 0700); err != nil {
+        return fmt.Errorf("failed to create .ssh directory: %w", err)
+    }
+
+    var knownHostsContent []string
+    if _, err := os.Stat(knownHostsPath); err == nil {
+        content, err := os.ReadFile(knownHostsPath)
+        if err != nil {
+            return fmt.Errorf("failed to read known_hosts file: %w", err)
+        }
+        knownHostsContent = strings.Split(string(content), "\n")
+    }
+
+    var output []byte
+    var scanErr error
+    for i := 0; i < 3; i++ {
+        output, scanErr = sshKeyscanCommand(targetAddress)
+        if scanErr == nil {
+            break
+        }
+        fmt.Printf("Attempt %d: failed to fetch host key for %s: %v\n", i+1, targetAddress, scanErr)
+        time.Sleep(time.Second)
+    }
+    if len(output) == 0 {
+        return fmt.Errorf("failed to fetch host key for %s after 3 attempts: last error: %w", targetAddress, scanErr)
```

Comment:
If a user fixes the issue with this, would they have to restart Alloy for things to start working? Ideally the component should just keep trying periodically until it works.